pkg/catalog/catalog.go

```diff
@@ -31,3 +33,28 @@ func NewMeshCatalog(computeInterface compute.Interface, certManager *certificate
 func (mc *MeshCatalog) GetTrustDomain() string {
 	return mc.certManager.GetTrustDomain()
 }
+
+func (mc *MeshCatalog) SpiffEnabled() bool {
```

Comment:
going to make this more similar to `GetTrustDomain` which has less chances for various types of failures after #5036

Reply:
updated after #5193